src/pages/Models/Details/ModelDetails.js

```diff
@@ -14,317 +15,94 @@ import { getModelUUID, getModelStatus } from "app/selectors";
 import { fetchModelStatus } from "juju/actions";
 import { collapsibleSidebar } from "app/actions";
 import {
-  generateStatusElement,
-  generateSpanClass,
-  generateIconPath
-} from "app/utils";
+  applicationTableHeaders,
+  unitTableHeaders,
+  machineTableHeaders,
+  relationTableHeaders,
+  generateApplicationRows,
+  generateMachineRows,
+  generateRelationRows,
+  generateUnitRows
+} from "./generators";
 
 import "./_model-details.scss";
 
-const applicationTableHeaders = [
-  { content: "app" },
-  { content: "status" },
-  { content: "version", className: "u-align--right" },
-  { content: "scale", className: "u-align--right" },
-  { content: "store" },
-  { content: "rev", className: "u-align--right" },
-  { content: "os" },
-  { content: "notes" }
-];
-
-const unitTableHeaders = [
-  { content: "unit" },
-  { content: "workload" },
-  { content: "agent" },
-  { content: "machine", className: "u-align--right" },
-  { content: "public address" },
-  { content: "port", className: "u-align--right" },
-  { content: "message" }
-];
-
-const machineTableHeaders = [
-  { content: "machine" },
-  { content: "state" },
-  { content: "az" },
-  { content: "instance id" },
-  { content: "message" }
-];
-
-const relationTableHeaders = [
-  { content: "relation provider" },
-  { content: "requirer" },
-  { content: "interface" },
-  { content: "type" },
-  { content: "message" }
-];
-
-const generateIconImg = (name, namespace) => {
-  return (
-    <img
-      alt={name + " icon"}
-      width="24"
-      height="24"
-      className="entity-icon"
-      src={generateIconPath(namespace)}
-    />
-  );
-};
-
-const generateEntityLink = (namespace, href, name, subordinate) => {
-  return (
-    <>
-      {subordinate && <span className="subordinate"></span>}
-      {namespace && generateIconImg(name, namespace)}
-      <a href={href}>{name}</a>
-    </>
-  );
-};
-
-const generateApplicationRows = modelStatusData => {
-  if (!modelStatusData) {
-    return [];
+/**
+  Returns the modelStatusData filtered by the supplied values.
+  @param {Object} modelStatusData The model status data to filter
+  @param {String} appName The name of the application to filter the data by.
+*/
+const filterModelStatusData = (modelStatusData, appName) => {
```

Comment:
This is the newly added function, unfortunately the diff makes it very hard to read, it might be best to view the new file instead.